frontend/server/server.ts

```diff
@@ -313,20 +313,43 @@ const createTensorboardHandler = async (req, res) => {
     return;
   }
   const logdir = decodeURIComponent(req.query.logdir);
+  const tfversion = decodeURIComponent(req.query.tfversion);
```

Comment:
You should first verify req.query.tfversion is non empty.
Because `decodeURIComponent(undefined) === 'undefined'`

Reply:
Will modify it.